File: LeetCode-August-Challenge/RomantoInteger.java
import java.util.*;

class Solution {
    public int romanToInt(String s) {
        HashMap<Character, Integer> mp = new HashMap<>();
        mp.put('V', 5);
        mp.put('I', 1);
        mp.put('X', 10);
        mp.put('L', 50);
        mp.put('C', 100);
        mp.put('D', 500);
        mp.put('M', 1000);

        int res = mp.get(s.charAt(s.length() - 1));

        for(int i = s.length() - 2; i >= 0; --i)
            res += mp.get(s.charAt(i)) < mp.get(s.charAt(i + 1)) ? -mp.get(s.charAt(i)) : +mp.get(s.charAt(i));
        return res;
    }
}

public class RomantoInteger {
    public static void main(String[] args) {
        String s = "MCMXCIV";
        Solution S = new Solution();
        System.out.println(S.romanToInt(s));
    }
}

File: LeetCode-August-Challenge/RomantoInteger.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int romanToInt(string s) {
        int values[100] = {0};
        values['I'] = 1;
        values['V'] = 5;
        values['X'] = 10;
        values['L'] = 50;
        values['C'] = 100;
        values['D'] = 500;
        values['M'] = 1000;

        int res = 0;
        for(int i = 0; i < s.size(); ++i)
            res += values[s[i]] < values[s[i + 1]] ? -values[s[i]] : +values[s[i]]; 
        return res;
    }
};

int main(){
    string s = "MCMXCIV";
    Solution S;
    cout<<S.romanToInt(s)<<endl;
    return 0;
}